packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -422,1422 +419,1101 @@ describe('NetworkController', () => {
+                              controller.getProviderAndBlockTracker().provider,
+                            );
+                          },
+                        },
+                      );
+
+                      await waitForStateChanges(messenger, {
+                        propertyPath: ['network'],
+                        count: 0,
+                        produceStateChanges: () => {
+                          controller
+                            .getProviderAndBlockTracker()
+                            .provider.emit('error', { some: 'error' });
+                        },
+                      });
+                      expect(controller.state.network).toBe('1');
                     },
-                  });
-                  expect(controller.state.network).toBe('1');
-                },
-              );
+                  );
+                });
+              });
             });
           });
-        });
-      });
 
-      describe('when the provider config in state contains a network type of "rpc"', () => {
-        describe('if the provider config contains an RPC target', () => {
-          it('sets the provider to a custom RPC provider initialized with the configured target, chain ID, nickname, and ticker', async () => {
-            await withController(
-              {
-                state: {
-                  providerConfig: {
-                    type: 'rpc',
-                    chainId: '123',
-                    nickname: 'some cool network',
-                    rpcTarget: 'http://example.com',
-                    ticker: 'ABC',
-                  },
-                },
-              },
-              async ({ controller }) => {
-                const fakeMetamaskProvider = buildFakeMetamaskProvider([
+          describe('when the provider config in state contains a network type of "rpc"', () => {
+            describe('if the provider config contains an RPC target', () => {
+              it('sets the provider to a custom RPC provider initialized with the configured target, chain ID, nickname, and ticker', async () => {
+                await withController(
                   {
-                    request: {
-                      method: 'eth_chainId',
-                    },
-                    response: {
-                      result: '0x1337',
+                    state: {
+                      providerConfig: {
+                        type: 'rpc',
+                        chainId: '123',
+                        nickname: 'some cool network',
+                        rpcTarget: 'http://example.com',
+                        ticker: 'ABC',
+                      },
                     },
                   },
-                ]);
-                createMetamaskProviderMock.mockReturnValue(
-                  fakeMetamaskProvider,
-                );
+                  async ({ controller }) => {
+                    const fakeJsonRpcClient = buildFakeInfuraClient();
```

Comment:
Hmm, this doesn't seem right to me, as we're not connecting to an Infura network in these tests. I would have expected `buildFakeJsonRpcClient` or something.

Reply:
unfortunately the tests are not doing what they appeared to be.